src/Shared/FileUtilities.cs

```diff
@@ -421,6 +421,72 @@ internal static string FixFilePath(string path)
         }
 
         /// <summary>
+        /// If on Unix, convert backslashes to slashes for strings that resemble paths.
+        /// The heuristic is if something resembles paths (contains slashes) check if the
+        /// first segment exists and is a directory.
+        /// Use a native shared method to massage file path. If the file is adjusted,
+        /// that qualifies is as a path.
+        /// </summary>
+        internal static string MaybeAdjustFilePath(string value)
```

Comment:
Possibly, but I wanted to emphasize that it's not just for the FileSystem and it does not process every string, only whatever looks like path (Maybe seems like a good indicator, but it may just be due to my lack of imagination). In addition to the file system related adjustment (slashes) it may also correct a path for a framework file on Mono.